File: src/main/resources/static/js/postFeedback.js
$(document).ready(function () {
    $('#assessmentForm').submit(function (event) {
        event.preventDefault();

        // Get form data
        let formData = {
            scoreDate: $('#scoreDate').val(),
            feedbackDate: $('#feedbackDate').val(),
            score: $('#score').val(),
            feedback: $('#feedback').val(),
            assessmentFeedbackReleased: $('#assessmentFeedbackReleased').is(":checked"),
            dateAssessmentFeedbackReleased: $('#dateAssessmentFeedbackReleased').val(),
            assessmentMarksEntered: $('#assessmentMarksEntered').is(":checked"),
            dateAssessmentMarksEntered: $('#dateAssessmentMarksEntered').val(),
            assessmentMarksTransferred: $('#assessmentMarksTransferred').is(":checked"),
            dateAssessmentMarksTransferred: $('#dateAssessmentMarksTransferred').val(),
            assessmentFeedbackEntered: $('#assessmentFeedbackEntered').is(":checked"),
            dateAssessmentFeedbackEntered: $('#dateAssessmentFeedbackEntered').val(),
            cohortFeedbackReturned: $('#cohortFeedbackReturned').is(":checked"),
            dateCohortFeedbackReturned: $('#dateCohortFeedbackReturned').val()
        };

        // Use AJAX to send data to the backend
        $.ajax({
            url: 'feedback',
            type: 'POST',
            contentType: 'application/json',
            data: JSON.stringify(formData),
            success: function (data) {
                console.log('Success:', data);
                alert('Form submitted successfully');
            },
            error: function (error) {
                console.error('Error:', error);
                alert('Error submitting form');
                // Handling error conditions
            }
        });
    });

    // Triggers an event when the button is clicked
    $('#calculateStatsBtn').click(function () {

        $.ajax({
            url: '/calculateStats',
            type: 'GET',
            success: function (data) {
                // Filling data into the dialog box
                $('#averageScore').text('Average Score: ' + data.averageScore);
                $('#minScore').text('Min Score: ' + data.minScore);
                $('#maxScore').text('Max Score: ' + data.maxScore);

                // Show dialog box
                $('#dialog').show();
            },
            error: function (error) {
                console.error('Error:', error);
                alert('Error calculating statistics');
            }
        });
    });

    // Close the dialog box when you click the close button
    $('#closeDialog').click(function () {
        $('#dialog').hide();
    });
});

File: src/main/java/com/assessment/comsc/feedbackform/PostFeedback.java
package com.assessment.comsc.feedbackform;

import lombok.Data;

import java.io.Serializable;
import java.util.Date;

@Data
public class PostFeedback implements Serializable {

    private Integer assessmentId;
    private Long userId;
    private Date scoreDate;
    private Date feedbackDate;
    private Double score;
    private String feedback;
    private Boolean assessmentMarksReleased;
    private Date dateAssessmentMarksReleased;
    private Boolean assessmentFeedbackReleased;
    private Date dateAssessmentFeedbackReleased;
    private Boolean assessmentMarksEntered;
    private Date dateAssessmentMarksEntered;
    private Boolean assessmentMarksTransferred;
    private Date dateAssessmentMarksTransferred;
    private Boolean assessmentFeedbackEntered;
    private Date dateAssessmentFeedbackEntered;
    private Boolean cohortFeedbackReturned;
    private Date dateCohortFeedbackReturned;
}

